### PR TITLE
spring task 7 attempt #1

### PR DESCRIPTION
**Pull request by AndriiBardin**

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: src/main/java/cinema/controller/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/controller/CustomGlobalExceptionHandler.java]: redundant empty line

File: src/main/java/cinema/controller/OrderController.java [hunk 2]
```diff
@@ -30,13 +31,17 @@ public OrderController(OrderService orderService, UserService userService,
...
+                .getByUser(userService.findByEmail(email).get()));
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
-        return orderService.getOrdersHistory(userService.get(userId)).stream()
+    public List<OrderResponseDto> getOrderHistory(Authentication authentication) {
+        String email = authentication.getName();
+        User user = userService.findByEmail(email).get();
+        return orderService.getOrdersHistory(userService.findByEmail(email).get()).stream()
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/controller/OrderController.java]: let's remove duplicated code

File: src/main/java/cinema/security/PasswordValidation.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+
+@Constraint(validatedBy = PasswordValidationImpl.class)
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface PasswordValidation {
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/security/PasswordValidation.java:14]: does it work without the required fields?

see your `EmailValidation` constraint

File: src/main/java/cinema/security/impl/PasswordValidationImpl.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package cinema.security.impl;
+
+import cinema.dto.UserRequestDto;
+import cinema.security.PasswordValidation;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class PasswordValidationImpl
+        implements ConstraintValidator<PasswordValidation, UserRequestDto> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/security/impl/PasswordValidationImpl.java:19]: redundant empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**